tips/034_gleaning.md

```diff
@@ -0,0 +1,252 @@
+## Effective AI Engineering #34: Gleaning
+
+**Does your AI give up after one failed attempt?** Your validator catches an error in the response, but instead of learning from the feedback, you start over from scratch and likely repeat the same mistake.
+
+Validation failures contain valuable information about what went wrong. Instead of discarding failed responses, you can feed the error details back to the AI and give it a chance to correct its mistakes iteratively.
+
+### The Problem
+
+Many developers treat validation failures as dead ends, regenerating responses from scratch without learning from errors. This creates challenges that aren't immediately obvious:
+
+```python
+# BEFORE: Discard failed responses and restart
```

Comment:
we should have a much simpler BEFORE example. We do NOT need anything but a simple AI call here.